lib/skate_web/controllers/detours_controller.ex

```diff
@@ -24,6 +24,24 @@ defmodule SkateWeb.DetoursController do
     json(conn, %{data: returned_uuid})
   end
 
+  @spec get_detours(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def get_detours(conn, _params) do
+    %{id: user_id} = AuthManager.Plug.current_resource(conn)
+
+    detours =
+      user_id
+      |> Detours.list_detours()
+      |> Enum.group_by(fn detour -> detour.status end)
```

Comment:
I think I'd prefer this to live in `&Detours.list_detours/1`. Moving it there would leave the controller to focus entirely on HTTP-layer stuff, rather than having to worry about figuring out which ones are `:draft`, etc.

Plus, right now, the `draft`/`active`/`past` trifurcation happens in the `Detours` module and then also a little bit here, and it would be nice to unify them.

I'll do some thinking about whether this is a blocking comment or a non-blocking nit.

Reply:
👍 That works for me. I reframed the function in Detours as `grouped_detours`